lmdeploy/messages.py

```diff
@@ -116,6 +92,12 @@ def __post_init__(self):
         assert self.top_p > 0 and self.top_p <= 1  # (0, 1]
         assert self.top_k >= 0, 'top_k can not be a negative integer'
         assert self.temperature >= 0 and self.temperature <= 2  # [0,2]
+        if self.do_sample is False:
+            self.top_k = 1
+            # top_k = 1 will do greedy decoding, set other params to default
+            # value to avoid unnecessary compute.
+            self.temperature = 1.0
```

Comment:
Users may not know the parameters have been changed.

Reply:
make sense